webapp/client/src/components/AnchorButton.test.js

```diff
@@ -0,0 +1,38 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import AnchorButton from "./AnchorButton";
+
+const MOCK_PROPS = {
+  text: "anchor text",
+  url: "/some/link/path",
+  isDownloadLink: false,
+};
+
+describe("AnchorButton", () => {
+  it("should render the text", () => {
+    render(<AnchorButton {...MOCK_PROPS} />);
+
+    expect(screen.getByText(MOCK_PROPS.text)).toBeInTheDocument();
+    expect(screen.getByText(MOCK_PROPS.text).closest("a")).toHaveAttribute(
+      "href",
+      expect.stringContaining(MOCK_PROPS.url)
+    );
+  });
+
+  it("should not has a download attribute", () => {
+    render(<AnchorButton {...MOCK_PROPS} />);
+
+    expect(screen.getByText(MOCK_PROPS.text).closest("a")).not.toHaveAttribute(
+      "download"
+    );
+  });
+
+  it("should has a download attribute", () => {
+    MOCK_PROPS.isDownloadLink = true;
```

Comment:
The problem with this approach is that you manipulating a key/value of the mock object (which is assumed to be a constant) and therefore tests are not independent and the running order becomes important. Maybe it is better to have something like this:
```js

const MOCK_PROPS = {
  text: "anchor text",
  url: "/some/link/path",
};
```

```js
it("should has a download attribute", () => {
    const PROPS_DATA = {...MOCK_PROPS, isDownloadLink: true}
    render(<AnchorButton {...PROPS_DATA} />);

    expect(screen.getByText(MOCK_PROPS.text).closest("a")).toHaveAttribute(
      "download"
    );
  });

  it("should not has a download attribute", () => {
    const PROPS_DATA = {...MOCK_PROPS, isDownloadLink: false}
    render(<AnchorButton {...PROPS_DATA} />);

    expect(screen.getByText(MOCK_PROPS.text).closest("a")).not.toHaveAttribute(
      "download"
    );
  });
```

Reply:
Thank you! And yes your totally right that this leads to not independent tests. :see_no_evil:

Reply:
Done!